lib/resty/core/request.lua

```diff
@@ -68,7 +68,7 @@ local truncated = ffi.new("int[1]")
 
 local req_headers_mt = {
     __index = function (tb, key)
-        return rawget(tb, (gsub(lower(key), '_', '-', "jo")))
+        return rawget(tb, (re.gsub(lower(key), '_', '-', "jo")))
```

Comment:
Alas. This is slower on all architectures. Here it relies on PCRE. Maybe we should introduce a C function to do this simple character replacement?

Reply:
@agentzh I just pushed a commit proposing an implementation. Do you think this new `str_replace_char` function should live in a different Lua module? `misc.lua` or a new `string.lua`? It isn't exposed as it stands and thus does not have tests yet. I will add them based on your feedback.

Reply:
@thibaultcha Could be in ` resty.core.base`.

Reply:
@thibaultcha Or `resty.core.util`. Maybe this is better.